engine/swarm/secrets.md

```diff
@@ -0,0 +1,351 @@
+    creates a mount point within the container with the same file name as the
+    flag's parameter.
+
+5.  Verify that the `mysql` service is running using `docker service ls`.
+
+    ```bash
+    $ docker service ls
+
+    ID            NAME   MODE        REPLICAS  IMAGE
+    6xw6rke8k1ul  mysql  replicated  1/1       mysql:latest
+    ```
+
+6.  Now that MySQL is set up, you can create a WordPress container that connects
+    to the MySQL server. For an overview of environment variables available
+    for configuring the `wordpress` image, see the
+    [wordpress image's documentation on Docker Hub](https://hub.docker.com/_/wordpress/).
+    
+    The WordPress service will have the following characteristics:
+
+    - Joins the service to the `wp_network` overlay network, and resolves to the
+      DNS name `wordpress`.
+    - Because the scale is set to 1, only a single WordPress task runs.
+      Load-balancing WordPress is left as an exercise to the reader, because of
+      limitations with storing WordPress session data on the container filesystem.
+    - Exposes WordPress on port 30000 of the host machine, so that you can access
+      it from external hosts. You could expose port 80 if you do not have a web
+      server running on port 80 already.
+    - Grants the service access to the `mysql_password` secret, but specify a
+      different target file name within the container. The WordPress container
+      will use the mount point `/run/secrets/wp_db_password`. Also specify that
+      the secret is not world-readable, by setting the mode to `0400`.
+    - Sets the environment variable `WORDPRESS_DB_PASSWORD_FILE` to this file
+      path. The `wordpress` service will read the password string from that file.
+
+    ```bash
+    $ docker service create --name="wordpress" \
+                            --network="wp_network" \
+                            --replicas="1" \
+                            --publish 30000:80 \
+                            --secret source=mysql_password,target=wp_db_password,mode=0400 \
+                            -e WORDPRESS_DB_PASSWORD_FILE="/run/secrets/wp_db_password" \
+                            -e WORDPRESS_DB_HOST="mysql:3306" \
+                            wordpress:latest
+    ```
+
+7.  Access `http://localhost:30000/` from any Swarm node and begin using
+    WordPress.
+
+## Example: Rotate a secret
+
+This example builds upon the previous one. In this scenario, you create a new
+secret with the new MySQL password, update the `mysql` and `wordpress` services
+to use it, then remove the old secret.
+
+1.  Create the new secret, and name it `mysql_password_v2`.
+
+    ```bash
+    $ openssl rand -base64 20 |docker secret create mysql_password_v2
+      lwxgks2rojdbvaui5kay1suvr
+    ```
+
+2.  Update the `mysql` service to use the new password, keeping the target path
+    at `/run/secrets/mysql_password` within the container.
+
+    ```bash
+    $ docker service update mysql \
```

Comment:
I think we should add an example of certificate rotation for an nginx service. It's a lot easier/more straightforward example. Alternatively, we can show rotation of a redis access database. Running MySQL database as a service is a hard sell. We might simply show it as a normal docker run, and manual update of the passphrase using `docker exec`, and rotation of the passphrase on the wordpress using `service update`